v2/cf-nodejs-logging-support/src/lib/logger/record-factory.ts

```diff
@@ -22,99 +32,190 @@ export default class RecordFactory {
         return RecordFactory.instance;
     }
 
+    updateCacheMsg() {
+        this.cacheMsgRecord = {};
+
+        const writtenAt = new Date();
+
+        const cachedFields = this.config.getCacheMsgFields();
+        let cache: any = {};
+        cachedFields.forEach(
+            field => {
+                if (!Array.isArray(field.source)) {
+                    cache[field.name] = this.sourceUtils.getFieldValue(field.source, cache, writtenAt);
+                } else {
+                    const value = this.sourceUtils.getValueFromSources(field, cache, "msg-log", writtenAt);
+
+                    if (value != null) {
+                        cache[field.name] = value;
+                    }
+                }
+
+                // Handle default
+                if (cache[field.name] == null && field.default != null) {
+                    cache[field.name] = field.default;
+                }
+
+                // Replaces all fields, which are marked to be reduced and do not equal to their default value to REDUCED_PLACEHOLDER.
+                if (field._meta!.isRedacted == true && cache[field.name] != null && cache[field.name] != field.default) {
+                    cache[field.name] = this.REDACTED_PLACEHOLDER;
+                }
+            }
+        );
+        this.cacheMsgRecord = cache;
+    }
```

Comment:
This is a clever approach to implement caching for fields. Thankfully, the cache can be invalidated and updated as a whole since we know which fields need to be cached based on the configuration. It avoids checking for cache misses over and over which really speeds up the libraries performance. Nice!

Ideas for improvement / thoughts
- I see the point why you decided to use two separate caches for msg and req logs. However, there is some code duplication here. Wouldn't it make sense to move the cache code to a separate class/file and make it reusable?